pkgs/by-name/re/regina-normal/package.nix

```
@@ -0,0 +1,120 @@
+{
+  lib,
+  stdenv,
+  hexdump,
+  locale,
+  fetchurl,
+  python3,
+  gnupg,
+  cmake,
+  doxygen,
+  gmp,
+  graphviz,
+  libxml2,
+  libxslt,
+  perl,
+  pkg-config,
+  qt6,
+  shared-mime-info,
+  lmdb,
+  databaseLibrary ? lmdb, # can be built with tokyocabinet instead
+  highDim ? false, # whether or not to include triangulations of dimension 9-15
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "regina-normal";
+  version = "7.4.1";
+
+  # Signed source tarball
+  src =
+    let
+      inherit (finalAttrs) version;
+      manifest = fetchurl {
+        url = "https://github.com/regina-normal/regina/releases/download/regina-${version}/SHA256SUMS-signed.txt";
+        hash = "sha256-uXbyHnmkosMg2/9ikNaN7vKA2IZSxSGg2TdimD/YsXU=";
+      };
+      # Benjamin Burton's PGP key
+      publicKey = fetchurl {
+        url = "https://keybase.io/benjaminburton/pgp_keys.asc";
+        hash = "sha256-uuFOj6O4Xz4SoxdqjpgaD6QZjfcyKkZFUQjBZnTGuSg=";
+      };
+    in
+    fetchurl {
+      url = "https://github.com/regina-normal/regina/releases/download/regina-${version}/regina-${version}.tar.gz";
+      hash = "sha256-dBMQFAmbdwg90dCQBP1ryXJCHHktGIzfQ/D9Ecpssu0=";
+      nativeBuildInputs = [ gnupg ];
+      downloadToTemp = true;
+      postFetch = ''
+        pushd "$(mktemp -d)" || exit 1
+        export GNUPGHOME=$PWD/gnupg
+        mkdir -m 700 "$GNUPGHOME"
+        ln -s ${manifest} SHA256SUMS-signed.txt
+        ln -s "$downloadedFile" regina-${version}.tar.gz
+        gpg --import ${publicKey}
+        gpg --trust-model always --verify SHA256SUMS-signed.txt
+        sha256sum -c --ignore-missing SHA256SUMS-signed.txt
+        popd || exit 1
+        mv "$downloadedFile" "$out"
+      '';
+    };
```

```suggestion
  src = fetchFromGitHub {
    owner = "regina-normal";
    repo = "regina";
    rev = "v${finalAttrs.version}";
    hash = "sha256-74131014099b77083dd1d09004fd6bc972421c792d188cdf43f0fd11ca6cb2ed";
  };
```

We can use the hash value from https://github.com/regina-normal/regina/releases/download/regina-7.4.1/SHA256SUMS-signed.txt directly (after verifying the PGP signature on our end).